crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -755,6 +829,48 @@ pub fn unitary_to_gate_sequence_inner(
         })
 }
 
+#[pyfunction]
+#[pyo3(signature = (unitary, target_basis_list, qubit, error_map=None, simplify=true, atol=None))]
+pub fn unitary_to_circuit(
+    py: Python,
+    unitary: PyReadonlyArray2<Complex64>,
+    target_basis_list: Vec<PyBackedStr>,
+    qubit: usize,
+    error_map: Option<&OneQubitGateErrorMap>,
+    simplify: bool,
+    atol: Option<f64>,
+) -> PyResult<Option<CircuitData>> {
+    let mut target_basis_vec: Vec<EulerBasis> = Vec::with_capacity(target_basis_list.len());
+    for basis in target_basis_list {
+        let basis_enum = EulerBasis::__new__(basis.deref())?;
```

Comment:
The `*basis` would end up in a copy I think but this does it all by reference. Although I'm not sure it would work because basis is a `PyBackedStr` so I don't think we can use `*basis`. `PyBackedStr.deref()`  returns a `&str` which is what we need in this case.

Reply:
I might be mistaken, but the docs suggest that `PyBackedStr` is meant to be used with deref coercion. You can use `__new__(basis)`.  [For example.](https://github.com/mtreinish/qiskit-core/compare/optimize-1q-more-rust...jlapeyre:qiskit-core:optimize-1q-more-rust-gjl-change)